velox/vector/ComplexVector.cpp

```diff
@@ -608,9 +608,31 @@ void ArrayVectorBase::copyRangesImpl(
 
 void RowVector::validate(const VectorValidateOptions& options) const {
   BaseVector::validate(options);
+  vector_size_t lastNonNullIndex{size()};
+
+  if (nulls_) {
+    lastNonNullIndex = bits::findLastBit(nulls_->as<uint64_t>(), 0, size());
+  }
+
   for (auto& child : children_) {
+    // TODO: Currently we arent checking for null children on ROWs
+    // since there are cases in SelectiveStructReader/DWIO/Koski where
+    // ROW Vectors with null children are created.
     if (child != nullptr) {
       child->validate(options);
+      if (child->size() < size()) {
+        // Cannot have null nulls buffer if child size < parent size.
```

Comment:
nit: comment not required as the error message below is proving enough context. Same for L629